### ISSUE TITLE
Options parsing: wrong bindings and incorrect arity of collection types

### ISSUE THREAD

**Issue by database64128** [dotnet/command-line-api#1359]
```cs
using System;
using System.CommandLine;
using System.CommandLine.Invocation;
using System.Text;

var pluginOption = new Option<string>("--plugin");
var pluginVersionOption = new Option<string>("--plugin-version");
var pluginWhateverOption = new Option<string[]>("--plugin-whatever");

var rootCommand = new RootCommand("")
{
    pluginOption,
    pluginVersionOption,
    pluginWhateverOption,
};

rootCommand.Handler = CommandHandler.Create<string, string, string, string[]>(
    (string plugin, string pluginVersion, string pluginOptions, string[] pluginWhatever) =>
    {
        Console.WriteLine($"Plugin: {plugin}");
        Console.WriteLine($"Plugin Version: {pluginVersion}");
        Console.WriteLine($"Plugin Whatever: {pluginWhatever.Length}");
    });

Console.OutputEncoding = Encoding.UTF8;
await rootCommand.InvokeAsync(args);
```

1. Passing `--plugin-version 1.0` causes `plugin` to become `1.0` too:

```
Plugin: 1.0
Plugin Version: 1.0
Plugin Whatever: 0
```

2. Passing `--plugin-whatever a b c d` is parsed as `[ --plugin-whatever <a> ]   ???--> b c d` instead of `[ --plugin-whatever <a> <b> <c> <d> ]`. Broken since #1339.

**Comment by jonsequitur** [comment]
This is an intended breaking change made in #1199.

You can revert to the old behavior by setting `pluginWhateverOption.AllowMultipleArgumentsPerToken = true`.

**Comment by database64128** [comment]
Fixed by #1361.
